### PR TITLE
add powershell script

### PR DESCRIPTION
**Pull request by arturogonzalez58**
This PR adds a new script to run in Windows and Powershell. We only support running the intermediate tour in the application using Linux and MacOs. This script is the first iteration to support Windows in the intermediate tour.

In order to test the pr you need the following:
- Windows(only windows 10 tested)
- kubectl
- Ability to use Powershell as an administrator
- Access to `app.getambassador.io`
- it is assuming that you doesn't have telepresence installed

Kown issues
- Telepresence is not updatable from the script
- If the user cancels the telepresence installation, the script is going to continue

Testing:
1. Go to app.getambassador.io and start the tour
image

2. Click on `Don't have a cluster? get a free one!`

4. Follow the instruction in the tour until step 3: Run the script
image

5. Using power shell as administrator move the branch of this PR:
```git checkout agv/dev/windows_script```

6. Then set the environment variable with the `AMBASSADOR_API_KEY` provided in the tour
```New-Item -Path Env:\AMBASSADOR_API_KEY -Value NDFSFADS....```

7.  Then run the script
```.\setup_dev_env.sp1```

### CODE REVIEW

File: emojivoto-web-app/js/setup_dev_env.ps1 [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+        if ((telepresence version) -match "upgrade") {
+            $install_telepresence = $true
+            Write-Host "Upgrading Telepresence"
+        } else {
+            Write-Host "Telepresence already installed"
+            send_telemetry("telepresenceAlreadyInstalled")
+        }
+    }
+    if ($install_telepresence) {
+        $telepresence_download_url = "https://app.getambassador.io/download/tel2/windows/amd64/latest/telepresence-setup.exe"
```
Comment (joseldmc) [inline emojivoto-web-app/js/setup_dev_env.ps1:150]: Let's target a specific version.

File: emojivoto-web-app/js/setup_dev_env.ps1 [hunk 2]
```diff
@@ -0,0 +1,211 @@
+param([string]$action="TP_SCIPT")
```
Comment (shepz) [inline emojivoto-web-app/js/setup_dev_env.ps1:1]: typo ?
```suggestion
param([string]$action="TP_SCRIPT")
```

File: emojivoto-web-app/js/setup_dev_env.ps1 [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+    Write-Host ''
+    Write-Host 'INSTRUCTIONS FOR DEVELOPMENT'
+    Write-Host '============================'
+    Write-Host 'To set the correct Kubernetes context on this shell, please execute:'
+    Write-Host 'export KUBECONFIG=./emojivoto_k8s_context.yaml'
+}
+
+use_telemetry
+has_cli
+set_os_arch
```
Comment (shepz) [inline emojivoto-web-app/js/setup_dev_env.ps1:202]: shouldn't we test the arch as a first step ?

File: emojivoto-web-app/js/setup_dev_env.ps1 [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+    }
+    
+}
+
+function check_init_config{
+    display_step(2)
+    Write-Host "Checking for AMBASSADOR_API_KEY environment variable"
+    if(-not ($Env:AMBASSADOR_API_KEY)){
+        # you will need to set the AMBASSADOR_API_KEY via the command line
+        # New-Item -Path Env:\AMBASSADOR_API_KEY -Value 'NTIyOWExZDktYTc5...'
```
Comment (shepz) [inline emojivoto-web-app/js/setup_dev_env.ps1:85]: remove comments ?

**Review by joseldmc — APPROVED** [review]

File: emojivoto-web-app/js/setup_dev_env.ps1 [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+}
+
+function connect_local_dev_env_to_remote{
+    $env:KUBECONFIG = './emojivoto_k8s_context.yaml'
+    display_step 5
+    Write-Host 'Connecting local development environment to remote K8s cluster'
+
+    $svcName = "edge-stack"
+    telepresence quit 2>&1 | Out-Null
+    telepresence helm upgrade --team-mode 2>&1 | Out-Null
```
Comment (knlambert) [inline emojivoto-web-app/js/setup_dev_env.ps1:164]: Is telepresence supposed to be already installed ?

File: emojivoto-web-app/js/setup_dev_env.ps1 [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+    telepresence login --apikey="$Env:AMBASSADOR_API_KEY"  2>&1 | Out-Null
+    telepresence quit -s 2>&1 | Out-Null
+    telepresence connect --docker 2>&1 | Out-Null
+
+    $interceptName = (kubectl get rs -n emojivoto --selector=app=web-app --no-headers -o custom-columns=":metadata.name")
+    telepresence intercept "$interceptName" --docker --context default -n "$Global:EMOJIVOTO_NS" --service web-app --port 8083:80 --ingress-port 80 --ingress-host "$svcName.ambassador" --ingress-l5 "$svcName.ambassador"
```
Comment (knlambert) [inline emojivoto-web-app/js/setup_dev_env.ps1:170]: This line just hang forever for me

image

**Review by knlambert — CHANGES_REQUESTED** [review]

File: emojivoto-web-app/js/setup_dev_env.ps1 (hunk 6 above)
Comment (knlambert) [reply emojivoto-web-app/js/setup_dev_env.ps1:170]: Do you have the issue ? Or is it a me problem ?

**Review by jacoblbeck — APPROVED** [review]

**Review by knlambert — APPROVED** [review]

**Pull request merged**